src/Raven.Server/Documents/Sharding/ShardedDocumentPutAction.cs

```diff
@@ -19,10 +18,11 @@ public ShardedDocumentPutAction(ShardedDocumentsStorage documentsStorage, Sharde
     protected override void ValidateId(Slice lowerId)
     {
         DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal, "Handle write documents to the wrong shard");
-        //var bucket = ShardHelper.GetBucketFromSlice(lowerId);
-        // throw if I'm not the owner of the bucket?
-        // throw if bucket moved, I'm the destination but still didn't confirm
-        // OR mark this document for migration
+        var config = _documentDatabase.ShardingConfiguration;
+        var bucket = ShardHelper.GetBucketFor(config, lowerId);
+        var shard = ShardHelper.GetShardNumberFor(config, bucket);
+        if (shard != _documentDatabase.ShardNumber)
+            throw new WrongShardException($"document '{lowerId}' was expected to be on shard #{shard}, but got to shard #{_documentDatabase.ShardNumber} (bucket: {bucket})");
```

Comment:
Can we fix the message? Typos, casing, having issues understanding the message, etc